content_sync/pipelines/definitions/concourse/common/steps.py

```diff
@@ -0,0 +1,248 @@
+import json
+from urllib.parse import urljoin
+
+from django.conf import settings
+from ol_concourse.lib.models.pipeline import (
+    Command,
+    DoStep,
+    GetStep,
+    Identifier,
+    PutStep,
+    Step,
+    StepModifierMixin,
+    TaskConfig,
+    TaskStep,
+    TryStep,
+)
+
+from content_sync.pipelines.definitions.concourse.common.identifiers import (
+    OCW_STUDIO_WEBHOOK_RESOURCE_TYPE_IDENTIFIER,
+    OPEN_DISCUSSIONS_RESOURCE_IDENTIFIER,
+    SLACK_ALERT_RESOURCE_IDENTIFIER,
+)
+from content_sync.pipelines.definitions.concourse.common.image_resources import (
+    CURL_REGISTRY_IMAGE,
+)
+from main.utils import is_dev
+
+
+def add_error_handling(
```

Comment:
Is there a reason to do the the OCW Studio Webhook  / Slack alert at the step-level instead of job-level?

I guess doing it at the step-level is pretty easy now, though.

**Also:** Should the typehint on step arg be `step: StepModifierMixin`?

Reply:
So, re: your first point, having error handlers at the step level lets us indicate more specifically about what failed in the Slack message. I suppose you're going to find out what went wrong pretty quickly once you pull up the pipeline in Concourse, but the issue remains that the URLs that the Slack alert shows by default don't properly link to our pipelines with the instance var in the query string, i.e.

https://cicd.odl.mit.edu/teams/ocw/pipelines/live/jobs/build-online-ocw-site/builds/2

vs where the link should go:

https://cicd.odl.mit.edu/teams/ocw/pipelines/live/jobs/build-online-ocw-site/builds/2?vars.site=%22matrix-calculus-for-machine-learning-and-beyond%22

What I was able to accomplish here is a link to the pipeline, but not the specific build, as that information is [intentionally withheld](https://github.com/concourse/concourse/issues/790) because they "don't want developers relying on the build number as a version number." Really, this is probably an issue with `concourse-slack-alert` itself and we should file it in their Github. If we set the alerts at the job level that would probably greatly reduce the size of the resulting JSON pipeline definition that is pushed up to Concourse, but the whole point of using this library is to obfuscate that part. We can neatly manage it in Python code, not so much just writing raw pipeline definitions.

As to the second point, I thought `Step` was more appropriate as that is what's logically being extended, but `StepModifierMixin` makes more sense.